cpp/velox/substrait/VeloxToSubstraitPlan.cc

```diff
@@ -242,10 +276,26 @@ void VeloxToSubstraitPlanConvertor::toSubstrait(
     // Process measure, eg:sum(a).
     ::substrait::AggregateFunction* aggFunction = aggMeasures->mutable_measure();
 
-    // Aggregation function name.
-    const auto& funName = aggregate.call->name();
-    // set aggFunction args.
+    // Use aggregate node's step information to write advanced extension 'allowIntermediateOutput'.
+    const auto& step = aggregateNode->step();
+    switch (step) {
+      case core::AggregationNode::Step::kPartial: {
+        substrait::extensions::AdvancedExtension ae{};
+        google::protobuf::StringValue msg;
+        msg.set_value("allowIntermediateOutput=1");
+        ae.mutable_optimization()->PackFrom(msg);
+        aggregateRel->mutable_advanced_extension()->MergeFrom(ae);
```

Comment:
Why don't we write this information to extension in Scala?

Reply:
`VeloxToSubstraitPlan.cc` this code is only used for testing. 

Scala part is [here](https://github.com/oap-project/gluten/pull/4130/files#diff-b40452a938d7e65cc63dfa2a22a8c2cf94044db05ed099496e6530972e3faa2cR178-R181)